src/main/java/com/esri/samples/na/closest_facility_static/ClosestFacilityStaticSample.java

```diff
@@ -0,0 +1,329 @@
+            try {
+              FeatureQueryResult facilitiesResult = result.get();
+
+              // add the found facilities to the list
+              for (Feature facilityFeature : facilitiesResult) {
+                facilitiesList.add(new Facility(facilityFeature.getGeometry().getExtent().getCenter()));
+              }
+
+            } catch (InterruptedException | ExecutionException e) {
+              displayMessage("Error retrieving list of facilities", e.getMessage());
+            }
+          });
+
+          // retrieve a list of all incidents
+          ListenableFuture<FeatureQueryResult> incidentsQueryResult = incidentsFeatureTable.queryFeaturesAsync(queryParameters);
+          incidentsQueryResult.addDoneListener(() -> {
+            try {
+              FeatureQueryResult incidentsResult = incidentsQueryResult.get();
+
+              // add the found incidents to the list
+              for (Feature incidentFeature : incidentsResult) {
+                incidentsList.add(new Incident(incidentFeature.getGeometry().getExtent().getCenter()));
+              }
+
+            } catch (InterruptedException | ExecutionException e) {
+              displayMessage("Error retrieving list of incidents", e.getMessage());
+            }
+          });
+
+          // enable the 'solve routes' button
+          solveRoutesButton.setDisable(false);
+
+          // resolve button press
+          solveRoutesButton.setOnAction(e -> {
+
+            // disable the 'solve routes' button and show the progress indicator
+            solveRoutesButton.setDisable(true);
+            progressIndicator.setVisible(true);
+
+            // start the routing task
+            closestFacilityTask.loadAsync();
+            closestFacilityTask.addDoneLoadingListener(() -> {
+              if (closestFacilityTask.getLoadStatus() == LoadStatus.LOADED) {
+                try {
+                  // create default parameters for the task and add facilities and incidents to parameters
+                  ClosestFacilityParameters closestFacilityParameters = closestFacilityTask.createDefaultParametersAsync().get();
+                  closestFacilityParameters.setFacilities(facilitiesList);
+                  closestFacilityParameters.setIncidents(incidentsList);
+
+                  // solve closest facilities
+                  try {
+                    // use the task to solve for the closest facility
+                    ListenableFuture<ClosestFacilityResult> closestFacilityTaskResult = closestFacilityTask.solveClosestFacilityAsync(closestFacilityParameters);
+                    closestFacilityTaskResult.addDoneListener(() -> {
+                      try {
+                        ClosestFacilityResult closestFacilityResult = closestFacilityTaskResult.get();
+
+                        // find the closest facility for each incident
+                        for (int indexOfIncident = 0; indexOfIncident < incidentsList.size(); indexOfIncident++) {
```

Comment:
indexOfIncident -> incidentIndex (this is what the argument is called in `getRankedFacilityIndexes`).